android-interop-testing/app/src/androidTest/java/io/grpc/android/integrationtest/TesterActivityTest.java

```diff
@@ -0,0 +1,58 @@
+package io.grpc.android.integrationtest;
+
+import android.os.Bundle;
+import android.test.ActivityInstrumentationTestCase2;
+import android.test.InstrumentationTestRunner;
+import junit.framework.Assert;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The Activity test of our App, used to trigger the integration test from commandline.
+ */
+public class TesterActivityTest extends ActivityInstrumentationTestCase2<TesterActivity> {
+
+  private String host;
+  private int port = 8030;
+  private String result;
+  private String serverHostOverride;
+  private String testCase;
+  private boolean useTls;
+  private boolean useTestCa;
+
+  public TesterActivityTest() {
+    super(TesterActivity.class);
+  }
+
+  public void setUp() throws Exception {
+    super.setUp();
+    getActivity();
+    Bundle args = ((InstrumentationTestRunner) getInstrumentation()).getArguments();
+    host = args.getString("server_host", "");
+    port = Integer.parseInt(args.getString("server_port", "0"));
+    assertTrue("Please specify server_host and server_port.", !host.isEmpty() && port != 0);
+    serverHostOverride = args.getString("server_host_override", null);
+    testCase = args.getString("test_case", "empty_unary");
+    useTls = Boolean.parseBoolean(args.getString("use_tls", "true"));
+    useTestCa = Boolean.parseBoolean(args.getString("use_test_ca", "false"));
+  }
+
+  public void testGrpc() throws Exception {
+    final CountDownLatch finished = new CountDownLatch(1);
+
+    new GrpcTestTask(testCase, host, port, serverHostOverride, useTls, useTestCa,
```

Comment:
It seems doing a instrumentation test doesn't gain us much, because we aren't really testing the activity and its interactions. Can we do a test that really doesn't use the UI much/at all? Maybe a ActivityUnitTestCase?